pkg/client/k8s/v1alpha1/resource.go

```diff
@@ -218,6 +219,16 @@ func (r *createOrUpdate) Apply(obj *unstructured.Unstructured, subresources ...s
 		}
 		return nil, err
 	}
+	// NOTE: We are returning installed unstructured object resources
+	// in case if they are already installed
+	if !canResourceOverWrite(resource.GetKind()) {
```

Comment:
Can we think more from the package p.o.v?
What is the purpose of `client/k8s/`?
Why should it bother about what should be applied/re-applied & what not?